packages/clay-dropdown/src/ClayDropdown.js

```diff
@@ -41,6 +41,15 @@ ClayDropdown.STATE = {
 	 */
 	expanded: Config.bool().value(false),
 
+	/**
+	 * Icon to use in the trigger.
+	 * @instance
+	 * @memberof ClayDropdown
+	 * @type {?string}
+	 * @default undefined
+	 */
+	icon: Config.string(),
```

Comment:
Why no using `symbol` and not `icon`?

Improve the API by passing `spritemap` into the `icon`. We generate a better consistency with the other components.
```javascript
icon: Config.shapeOf({
    spritemap: Config.string().required(),
    symbol: Config.string().value('caret-bottom')
}).required(),
```

Soy using:
```soy
{if $icon.symbol != 'caret-bottom'}
	{call ClayIcon.render}
	    {param spritemap: $icon.spritemap /}
            {param symbol: $icon.symbol /}
	{/call}
{/if}

{if $label}
	{$label}
{/if}

{if $icon.symbol == 'caret-bottom'}
	{call ClayIcon.render}
		{param spritemap: $icon.spritemap /}
		{param symbol: $icon.symbol /}
	{/call}
{/if}
```

Would it be less confusing what do you think?

Reply:
For some reason, the `symbol` is not set to the default value when it is not passed.

Reply:
Hey @matuzalemsteles, this attempts to achieve exactly that. Check https://github.com/brianchandotcom/liferay-portal/pull/53237 to see how the components are used in the end. 

Having complex state hierarchies makes them way too difficult to use outside our simple scenario. We need to aim for more flatten states and also remove prefixes as much as possible so the APIs are simpler. For instance, a dropdown will always show a `label` and an `icon`, no need to call that `triggerLabel` and `triggerIcon.symbol`.

Reply:
Yeah, It's very more complexity. 😅